### PR TITLE
	Added storable function class

### PR DESCRIPTION
**Pull request by R3AL**

### CODE REVIEW

File: include/rpc/function_proxy.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#include <string>
+#include <future>
+
+#include "msgpack.hpp"
+
+namespace rpc {
+
+class client;
+
+namespace call_type {
```
Comment (sztomi) [inline include/rpc/function_proxy.h]: Would it be possible for `get_function` to return a functor that knows if it is a sync or async function proxy?

**Comment by sztomi** [comment]
Thanks! Could you:

  * create the pull request against the dev branch instead of master
  * write unit tests that cover the new code (see coveralls)
  * use clang-format

?

File: include/rpc/function_proxy.h (hunk 1 above)
Comment (R3AL) [inline include/rpc/function_proxy.h]: Yes, I will change soon

Comment (sztomi) [inline include/rpc/function_proxy.h]: @R3AL great, thanks. Please bear with me as I'm going on a holiday for a few days, so I'll only be able to respond then.

**Comment by sztomi** [comment]
@R3AL I'm going to close this due to no activity.

**Pull request closed without merging**